### DISCUSSION TITLE
Dynamic Chart

### DISCUSSION THREAD

**Discussion by timuebbing** [KoalaPlot/koalaplot-core#59] [Q&A] [answered]
Hello, I am new to programming with kotlin and would like to create a desktop app that displays a graph. It should be possible to update this graph by pressing a button. How do I make this work?

**Comment by gsteckman** [comment, marked as answer]
This is definitely possible but if you are new to programming and to Compose I suggest you first take the time to go through the Compose documentation and tutorials to learn how to make applications that can respond to user input (assuming you have already gone through the Kotlin learning process). Then it is only a matter of changing the state that is used to generate the graph. The [PieChart sample](https://github.com/KoalaPlot/koalaplot-samples/blob/main/src/commonMain/kotlin/io/github/koalaplot/sample/PieSample.kt) is one example of how to change plot properties based on user settings.

**Reply by timuebbing** [reply]
Thanks for the quick reply. I will take a closer look at the things you mentioned.
